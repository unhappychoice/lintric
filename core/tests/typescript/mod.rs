use insta::assert_snapshot;
use lintric_core::analyze_code;
use serde_json;
use std::env;

#[test]
fn test_analyze_code_typescript() {
    let file_path = "tests/typescript/fixtures/basic_typescript_code.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "typescript_basic_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "typescript_basic_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_class_method_dependency() {
    let file_path = "tests/typescript/fixtures/class_method_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "class_method_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "class_method_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_import_dependency() {
    let file_path = "tests/typescript/fixtures/import_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "import_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "import_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_function_parameter_dependency() {
    let file_path = "tests/typescript/fixtures/function_parameter_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "function_parameter_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "function_parameter_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}

#[test]
fn test_typescript_arrow_function_parameter_dependency() {
    let file_path = "tests/typescript/fixtures/arrow_function_parameter_dependency.ts".to_string();
    let (ir, result) = analyze_code(file_path.clone()).unwrap();

    assert_snapshot!(
        "arrow_function_parameter_dependency_ir",
        serde_json::to_string_pretty(&ir).unwrap()
    );
    assert_snapshot!(
        "arrow_function_parameter_dependency_metrics",
        serde_json::to_string_pretty(&result).unwrap()
    );
}
