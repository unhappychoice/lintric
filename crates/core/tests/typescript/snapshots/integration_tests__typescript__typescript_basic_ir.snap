---
source: core/tests/typescript/mod.rs
expression: "serde_json::to_string_pretty(&ir).unwrap()"
---
{
  "file_path": "tests/typescript/fixtures/basic_typescript_code.ts",
  "definitions": [
    {
      "name": "x",
      "position": {
        "start_line": 1,
        "start_column": 7,
        "end_line": 1,
        "end_column": 8
      },
      "definition_type": "VariableDefinition"
    },
    {
      "name": "y",
      "position": {
        "start_line": 2,
        "start_column": 5,
        "end_line": 2,
        "end_column": 6
      },
      "definition_type": "VariableDefinition"
    },
    {
      "name": "foo",
      "position": {
        "start_line": 3,
        "start_column": 10,
        "end_line": 3,
        "end_column": 13
      },
      "definition_type": "FunctionDefinition"
    }
  ],
  "dependencies": [
    {
      "source_line": 2,
      "target_line": 1,
      "symbol": "x",
      "dependency_type": "VariableUse",
      "context": "variable_use"
    },
    {
      "source_line": 4,
      "target_line": 2,
      "symbol": "y",
      "dependency_type": "VariableUse",
      "context": "variable_use"
    }
  ],
  "usage": [
    {
      "name": "x",
      "kind": "Identifier",
      "position": {
        "start_line": 2,
        "start_column": 9,
        "end_line": 2,
        "end_column": 10
      }
    },
    {
      "name": "y",
      "kind": "Identifier",
      "position": {
        "start_line": 4,
        "start_column": 12,
        "end_line": 4,
        "end_column": 13
      }
    }
  ],
  "analysis_metadata": {
    "language": "TypeScript",
    "total_lines": 5,
    "analysis_timestamp": "now",
    "lintric_version": "0.1.0"
  }
}
